.github/workflows/test.yaml

```diff
@@ -0,0 +1,33 @@
+name: test
+
+on: [push]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [3.6, 3.7, 3.8, 3.9]
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install pytest
+          if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
+
+      - name: Install psf-weather-station
+        run: |
+          python setup.py install
+
+      - name: Test with pytest
+        run: |
+          pytest
```

Comment:
The test suite looks like it probably has pretty good coverage.  As you move this towards a 1.0 release, you might want to include codecov here.  TreeCorr has a relatively clean example you can follow to see how I run it in CI and upload it: https://github.com/rmjarvis/TreeCorr/blob/releases/4.3/.github/workflows/ci.yml#L137

You also need to connect your GitHub account to [CodeCov](https://about.codecov.io/), and I think you need a `.codecov.yml` file, cf. https://github.com/rmjarvis/TreeCorr/blob/releases/4.3/.codecov.yml  (Maybe not -- that might now be doable all in the codecov website -- I'm not sure.)